### PR TITLE
Fixing fantasy cards, trading items and make account transforms more resilient

### PR DESCRIPTION
**Pull request by Crazy-Duck**

### CODE REVIEW

File: handlers/cache.js [hunk 1]
```diff
@@ -30,15 +29,54 @@ var cacheTypeIDs = {
...
+            break;
+        case cacheTypeIDs.CSODOTALobbyInvite:
+            this.Logger.debug("Lobby invite cleared");
+            this.LobbyInvite = null;
+            this.emit("lobbyInviteCleared");
+            break;
+        case cacheTypeIDs.CSOEconItem:
+            this.Logger.debug("Traded item away");
+            var item = Dota2.schema.lookupType("CSOEconItem").decode(object_data);
+            this.Inventory = this.Inventory.filter(i => item.id.equals(i.id));
```
Comment (Ne3tCode) [inline handlers/cache.js]: Use `notEquals` (or `! equals`) if you want to remove a _single_ item from an array.

In addition:
> Note that the `Inventory` property will be the old value until after this event completes to allow comparison between the two.

So you probably have to call `emit` _before_ `filter`.

**Comment by Crazy-Duck** [comment]
Fix #519

**Comment by Crazy-Duck** [comment]
Fix #520

**Comment by Crazy-Duck** [comment]
Fix #516

**Pull request merged**